diff -ur ffi-yajl-2.3.1.orig/ext/ffi_yajl/ext/encoder/extconf.rb ffi-yajl-2.3.1/ext/ffi_yajl/ext/encoder/extconf.rb
--- ffi-yajl-2.3.1.orig/ext/ffi_yajl/ext/encoder/extconf.rb	2026-03-20 15:23:36.892973559 +0100
+++ ffi-yajl-2.3.1/ext/ffi_yajl/ext/encoder/extconf.rb	2026-03-20 15:24:49.994270674 +0100
@@ -1,13 +1,10 @@
 # rubocop:disable Style/GlobalVars
 require "mkmf"
 require "rubygems"
-require "libyajl2"
 
 RbConfig::MAKEFILE_CONFIG["CC"] = ENV["CC"] if ENV["CC"]
 
 # pick up the vendored libyajl2 out of the libyajl2 gem
-$CFLAGS = " -I#{Libyajl2.include_path} #{$CFLAGS}"
-$LDFLAGS = " -L#{Libyajl2.opt_path} #{$LDFLAGS}"
 
 # remove "-Wl,--no-undefined" flag if existent to allow for loading with dlopen
 $LDFLAGS.slice!("-Wl,--no-undefined")
diff -ur ffi-yajl-2.3.1.orig/ext/ffi_yajl/ext/parser/extconf.rb ffi-yajl-2.3.1/ext/ffi_yajl/ext/parser/extconf.rb
--- ffi-yajl-2.3.1.orig/ext/ffi_yajl/ext/parser/extconf.rb	2026-03-20 15:23:36.892664799 +0100
+++ ffi-yajl-2.3.1/ext/ffi_yajl/ext/parser/extconf.rb	2026-03-20 15:24:49.996378093 +0100
@@ -1,13 +1,10 @@
 # rubocop:disable Style/GlobalVars
 require "mkmf"
 require "rubygems"
-require "libyajl2"
 
 RbConfig::MAKEFILE_CONFIG["CC"] = ENV["CC"] if ENV["CC"]
 
 # pick up the vendored libyajl2 out of the libyajl2 gem
-$CFLAGS = "-I#{Libyajl2.include_path} #{$CFLAGS}"
-$LDFLAGS = "-L#{Libyajl2.opt_path} #{$LDFLAGS}"
 
 # remove "-Wl,--no-undefined" flag if existent to allow for loading with dlopen
 $LDFLAGS.slice!("-Wl,--no-undefined")
diff -ur ffi-yajl-2.3.1.orig/lib/ffi_yajl/ext.rb ffi-yajl-2.3.1/lib/ffi_yajl/ext.rb
--- ffi-yajl-2.3.1.orig/lib/ffi_yajl/ext.rb	2026-03-20 15:23:36.890821214 +0100
+++ ffi-yajl-2.3.1/lib/ffi_yajl/ext.rb	2026-03-20 15:24:49.992021692 +0100
@@ -20,7 +20,6 @@
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-require "rubygems"
 
 require "ffi_yajl/encoder"
 require "ffi_yajl/parser"
diff -ur ffi-yajl-2.3.1.orig/lib/ffi_yajl/ffi.rb ffi-yajl-2.3.1/lib/ffi_yajl/ffi.rb
--- ffi-yajl-2.3.1.orig/lib/ffi_yajl/ffi.rb	2026-03-20 15:23:36.890821214 +0100
+++ ffi-yajl-2.3.1/lib/ffi_yajl/ffi.rb	2026-03-20 15:24:49.989755059 +0100
@@ -20,9 +20,7 @@
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-require "rubygems"
 
-require "libyajl2"
 begin
   require "ffi"
 rescue LoadError
diff -ur ffi-yajl-2.3.1.orig/lib/ffi_yajl/map_library_name.rb ffi-yajl-2.3.1/lib/ffi_yajl/map_library_name.rb
--- ffi-yajl-2.3.1.orig/lib/ffi_yajl/map_library_name.rb	2026-03-20 15:23:36.891121200 +0100
+++ ffi-yajl-2.3.1/lib/ffi_yajl/map_library_name.rb	2026-03-20 15:24:44.409825582 +0100
@@ -20,7 +20,6 @@
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-require "libyajl2"
 
 # Mixin for use in finding the right yajl library on the system.  The 'caller'
 # needs to also mixin either the FFI module or the DLopen module.  Those are
@@ -45,16 +44,7 @@
     # @api private
     # @return Array<String> Array of yajl library names for platform
     def library_names
-      case host_os
-      when /mingw|mswin/
-        [ "libyajl.so", "yajl.dll" ]
-      when /cygwin/
-        [ "libyajl.so", "cygyajl.dll" ]
-      when /darwin/
-        [ "libyajl.bundle", "libyajl.dylib" ]
-      else
-        [ "libyajl.so" ]
-      end
+      [ "libyajl.so.2" ]
     end
 
     # Array of yajl library names prepended with the libyajl2 path to use to
@@ -71,14 +61,9 @@
       end.compact
     end
 
-    # Iterate across the expanded library names in the libyajl2-gem and then
-    # attempt to load the system libraries.  Uses the native dlopen extension
-    # that ships in this gem.
-    #
-    # @api private
     def dlopen_yajl_library
       found = false
-      ( expanded_library_names + library_names ).each do |libname|
+      ( library_names ).each do |libname|
         begin
           dlopen(libname)
           found = true
